### PR TITLE
master branch updates

### PR DESCRIPTION
**Pull request by mike-scott**

### CODE REVIEW

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Succeeded.

**Comment by gatek33p3r** [comment]
Nitrogen Deployment Failed.

**Comment by gatek33p3r** [comment]
K64F Deployment Failed.

**Comment by rsalveti** [comment]
LGTM, but not tested.

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
K64F Deployment Failed.

**Comment by mike-scott** [comment]
I successfully tested this PR today (with linaro/zephyr master-upstream-dev branch) for Carbon, Nitrogen and K64F.  However, I have yet to do full roll outs.  I'll add another comment once I've done that.

**Comment by gatek33p3r** [comment]
Nitrogen Deployment Succeeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Succeeded.

File: src/main.c [hunk 1]
```diff
@@ -252,16 +253,19 @@ static void bluemix_service(void)
...
-			/* restart the failed attempt counter */
-			bluemix_failures = 0;
-		} else {
-			bluemix_failures++;
-			OTA_DBG("Failed Bluemix init - attempt %d\n\n\n",
-				bluemix_failures);
-			tcp_interface_unlock();
-			continue;
+		if (!bluemix_inited) {
+			ret = bluemix_init(&bluemix_context);
```
Comment (mbolivar) [inline src/main.c]: Probably want to turn MQTT keepalive back on and send pings somewhere in this series, too.

Comment (mike-scott) [inline src/main.c]: Ah right!  Let me re-add.

Comment (mike-scott) [inline src/main.c]: Hrm.. I think we needed to re-work how MQTT packets were handled in the Zephyr lib code before we can re-enable the ping functions.  Probably have to address that in a new patchset.

**Comment by mike-scott** [comment]
I tested 3 rollouts to my 12 devices (3x Nitrogen, 8x Carbon and 1x K64F) successfully.   I have some future changes incoming for Kconfig support and will probably enable DHCPv4 (vs static MY IP) for K64F, but I think this patchset is good.

**Comment by mbolivar** [comment]
LGTM; tested on all three IoT devices using HiKey build 52.

**Pull request merged**